src/main/java/org/apache/sling/scripting/core/impl/DefaultSlingScript.java

```diff
@@ -742,13 +746,20 @@ Bindings verifySlingBindings(final SlingBindings slingBindings) throws IOExcepti
                 LOGGER.trace("Invoking addBindings() of {} took {} nanoseconds",
                         provider.getClass().getName(), stop-start);
                 if (stop-start > WARN_LIMIT_FOR_BVP_NANOS) {
-                    LOGGER.info("Adding the bindings of {} took {} microseconds which is above the harcoded limit of {} microseconds;"
-                            + " if this message appears often it indicates that this BindingsValuesProvider has an impact on general page rendering performance",
-                            new Object[]{provider.getClass().getName(), (stop-start)/1000, WARN_LIMIT_FOR_BVP_NANOS/1000});
+                    // SLING-11182 - make this work with older implementations of the Sling API
+                    if (request != null && request.getRequestProgressTracker() != null) {
+                        request.getRequestProgressTracker().log(String.format(BINDINGS_THRESHOLD_MESSAGE, provider.getClass().getName(), (stop-start)/1000, WARN_LIMIT_FOR_BVP_NANOS/1000));
```

Comment:
in case a progressTracker is available it just writes the message into the RPT, but not anymore into the logs (as before).
Technically that would count as regression, but I was never really confident with this way of reporting.

Reply:
I know, but it was a conscious decision - I'd like this logging to be consistent. :D